File: app/lib/models/guild_config.dart
class GuildConfig {
  final String guildId;
  Set<String> commandChannelIds;
  String? autoJoinAudioChannelId;
  String? followedUserId;

  GuildConfig({
    required this.guildId,
    Set<String>? commandChannelIds,
    this.autoJoinAudioChannelId,
    this.followedUserId,
  }) : commandChannelIds = commandChannelIds ?? {};

  factory GuildConfig.fromJson(Map<String, dynamic> json) => GuildConfig(
        guildId: json['guildId'] as String,
        commandChannelIds: ((json['commandChannelIds'] as List<dynamic>?) ?? [])
            .map((e) => e as String)
            .toSet(),
        autoJoinAudioChannelId: json['autoJoinAudioChannelId'] as String?,
        followedUserId: json['followedUserId'] as String?,
      );

  Map<String, dynamic> toJson() => {
        'guildId': guildId,
        'commandChannelIds': commandChannelIds.toList(),
        'autoJoinAudioChannelId': autoJoinAudioChannelId,
        'followedUserId': followedUserId,
      };
}

File: worker/Models/GuildConfig.cs
using System.Text.Json.Serialization;

namespace CordCastWorker.Models;

public class GuildConfig
{
    [JsonPropertyName("guildId")]
    public string GuildId { get; set; } = "";

    [JsonPropertyName("commandChannelIds")]
    public HashSet<string> CommandChannelIds { get; set; } = new();

    [JsonPropertyName("autoJoinAudioChannelId")]
    public string? AutoJoinAudioChannelId { get; set; }

    [JsonPropertyName("followedUserId")]
    public string? FollowedUserId { get; set; }

    public bool IsCommandAllowed(string channelId) =>
        CommandChannelIds.Count == 0 || CommandChannelIds.Contains(channelId);
}
